### PR TITLE
mtest: add additional test cases

### PR DESCRIPTION
**Pull request by pavanbalaji**
## Pull Request Description

Our testing seems to have missed some test cases that show up in real
applications.  This extended testsuite should cover at least some of
those cases.

## Expected Performance Changes

None.

## Known Issues

None.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Passes tests (included warning check)
* [x] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [x] Commits are self-contained and do not do two things at once
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by hzhou** [comment]
test:mpich/custom

testlist: pt2pt/pingping 2 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-sendcnt=65530 arg=-recvcnt=65530 arg=-seed=41 arg=-testsize=8
testlist: pt2pt/pingping 2 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-sendcnt=65530 arg=-recvcnt=131060 arg=-seed=42 arg=-testsize=8
testlist: pt2pt/sendrecv1 2 arg=-type=MPI_INT arg=-sendcnt=65530 arg=-recvcnt=65530 arg=-seed=59 arg=-testsize=32
testlist: pt2pt/sendrecv1 2 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-sendcnt=65530 arg=-recvcnt=65530 arg=-seed=65 arg=-testsize=32
testlist: pt2pt/sendrecv1 2 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-sendcnt=65530 arg=-recvcnt=131060 arg=-seed=66 arg=-testsize=32
testlist: rma/accpscw1 4 arg=-type=MPI_INT arg=-count=65530 arg=-seed=111 arg=-testsize=16
testlist: rma/epochtest 4 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-count=65530 arg=-seed=126 arg=-testsize=16
testlist: rma/getfence1 2 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-count=65530 arg=-seed=138 arg=-testsize=16
testlist: rma/lockall_dt 4 arg=-type=MPI_INT arg=-count=65530 arg=-seed=197 arg=-testsize=16
testlist: rma/putfence1 2 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-count=65530 arg=-seed=260 arg=-testsize=16
testlist: rma/putpscw1 4 arg=-type=MPI_INT:4+MPI_DOUBLE:8 arg=-count=65530 arg=-seed=274 arg=-testsize=16

testlist: rma/lockall_dt_flushlocalall 4 arg=-type=MPI_INT arg=-count=65530 arg=-seed=245 arg=-testsize=16

File: test/mpi/dtpools/src/dtpools_attr.c [hunk 1]
```diff
@@ -24,6 +24,33 @@
...
+        }
+    }
+
+    for (int i = 0; i < numblks; i++)
+        blklens[i] = 0;
+
+    for (int i = 0; i < total_blklen; i++)
+        blklens[DTPI_rand(dtpi) % numblks]++;
+}
+
```
Comment (hzhou) [inline test/mpi/dtpools/src/dtpools_attr.c]: @pavanbalaji This algorithm is different from generate `numblks-1` dividers directly. By randomly add 1 to one of the block, the probability of have a 0 block is `[(numblks-1)/numblks]^total_blklen`, even with 10 blocks and 100 total, `0.9^100` is `0.000026`, i.e. very small. By randomly picking `numblks-1` divider, according to https://en.wikipedia.org/wiki/Birthday_problem, is `~0.39`, i.e. very significant. When `total_blklen` is very big, your algorithm will result in very uniform blocks, but the other algorithm will result in quite "random" block sizes.

Comment (hzhou) [reply test/mpi/dtpools/src/dtpools_attr.c]: Talking to some friend and here is another algorithm: simply generate `numblks` random numbers then scale down (or up) to the total. The last one may need padding of plus or minus one.

With this algorithm, the distribution of each `blklen` is uniform (from 0 to the scaled upperbound). In the case of randomly adding `1`, the distribution of each `blklen` is gaussian. I am not exactly sure yet what is the distribution randomly picking dividers would result.

**Comment by pavanbalaji** [comment]
Yes, I know.  I wasn’t looking for a review yet.

**Pull request closed without merging**